tests/leakage_gadget_test.py

```diff
@@ -69,13 +71,15 @@ def test_compare_with_prune() -> None:
             ),
             n_shots=detection_circuit_shots.Shots,
         )
-        for detection_circuit_shots in detection_circuit_shots_list
+        for detection_circuit_shots in cast(
+            list[CircuitShots], detection_circuit_shots_list
+        )
     ]
 
     qermit_result_list = postselection_task(
         (
             result_list,
-            postselect_mgr_list,
+            postselect_mgr_list,  # type: ignore
```

Comment:
This is required since PostselectMgr, is not a recognised wire type. Simply adding it to the list of recognised wire types induces cyclic imports.

I don't really know what the right solution to this is...

I assume there must be a way to define a type annotation for PostselectMgr within `mittask.py` although I have tried and failed to do that. User defined generic types might also help -> https://docs.python.org/3/library/typing.html#user-defined-generic-types but I think that only works for python 3.12. If you happen to know how to do something like that then I would be very interested to know about that.

Another solution is to move serialised versions of PostselectMgr along wires. As far as I can tell that is the best solution, but would be some significant work.

Another solution would be to allow Any as a wire type, which I don't want to do...